File: generators/maa2java-xtend/src/test/resources/models/initial/genTest/SubCompImpl.java
package genTest;

import de.montiarcautomaton.runtimes.timesync.implementation.IComputable;
import java.lang.*;
import java.util.*;

class SubCompImpl     
implements IComputable<SubCompInput, SubCompResult> {

  public SubCompImpl() {
  }

  public SubCompResult getInitialValues() {
	  return new SubCompResult();
  }
 public SubCompResult compute(SubCompInput input) {
	 System.out.println("Testing");
	 return new SubCompResult();
 }

}

File: generators/montithings2cpp/src/test/resources/hwc/ipcTest/SubCompImpl.cpp
#include <iostream>
#include "SubCompImpl.h"

SubCompResult SubCompImpl::getInitialValues(){
	return SubCompResult();
	
}

SubCompResult SubCompImpl::compute(SubCompInput input){
    SubCompResult result;
    if (input.getInPort()) {
        auto arr = input.getInPort().value();
        std::cout << arr[0];
        result.setOutPort(arr[0]);

    }
    return result;
}